### DISCUSSION TITLE
How to implement a backend extension execution of CPU-consuming operations without affecting the front-end UI interaction blocking problem？

### DISCUSSION THREAD

**Discussion by zhaomenghuan** [eclipse-theia/theia#8537] [General] [answered]
When I use the backend extension to call the node module to compile my project code, during the compilation process, I found that I was operating the file tree at this time, and found that only loading did not respond quickly.

<img width="250" alt="屏幕快照 2020-09-21 下午7 46 30" src="https://user-images.githubusercontent.com/13075561/93762982-2b702680-fc43-11ea-9612-409c5e40e50d.png">

**Comment by vince-fugnitto** [comment]
@zhaomenghuan I cannot be certain without looking at the source code of your changes, but you should look towards making the compilation of the project `asynchronous` as to not block other processes.

**Comment by zhaomenghuan** [comment, marked as answer]
Finally, I checked that it had nothing to do with the front-end and back-end communication layer of Theia. The main reason was that the main Node.js thread was occupied. This problem was solved through the worker_threads module in the Node service.
